### PR TITLE
Improve developer documentation

### PR DESCRIPTION
**Pull request by nareddyt**
Mostly moving content around and adding new content. Diff looks bad but not much content was deleted.

- Move developer docs into `DEVELOPER.md` with clear instructions on prereqs, running the functions, and testing.
- In `DEVELOPER.md`, remove the instructions to run the app function manually. Only document the docker-based approach now.
- Move implementation details about each function into their own folders.
- Check in sample payload for testability.
- Check in `Pipfile` for easy virtualenv installation with specific python version.

### CODE REVIEW

**Comment by nareddyt** [comment]
Please take a look

File: blob-parser/readme.md [hunk 1]
```diff
@@ -1,11 +1,14 @@
-# BlobTrigger - Python
+# Blob Parser
 
-The `BlobTrigger` makes it incredibly easy to react to new Blobs inside of Azure Blob Storage. This sample demonstrates a simple use case of processing data from a given Blob using Python.
+Blob parser parses the HTML from the Blob Storage into JSON for the cosmos DB.
+It then inserts the data into the DB.
 
-## How it works
+Note that this function produces a lot of output after starting the function app, because it is continuously checking for its trigger. For this reason, you may want to disable this function during local development if it's not needed. To do this, click the 'A' in VS Code sidebar, find the function at the bottom under Workspace > Local Project > Functions, right-click on it and click Disable. You can use the same menu to re-enable it when you need it again.
```
Comment (newswim) [inline blob-parser/readme.md:6]: Do we mention installing the Azure extension for VS Code? If not, this might be a good place to mention it.

**Review by newswim — APPROVED** [review]
These are fantastic changes, thank you Teju!

One thing came to mind as I was reading -- I wonder if the M1 workaround using Docker is still necessary. Some time has passed since @DevTrav first ran into those issues, and I wonder if MS has added support. That might be a good task for someone to check out at the next working session. I can make a follow-up issue.

I had one other question about dev setup, but it's not a blocker, see comment.

**Comment by newswim** [comment]
Added ticket to investigate support for M1 chips: https://github.com/open-austin/azure-indigent-defense/issues/56

**Comment by nareddyt** [comment]
> Added ticket to investigate support for M1 chips: #56

Thanks for the review! Responded inline in the issue

File: blob-parser/readme.md (hunk 1 above)
Comment (nareddyt) [reply blob-parser/readme.md]: Good catch. Added

**Pull request merged**